src/SpendPermissionManager.sol

```diff
@@ -14,6 +14,24 @@ import {EIP712} from "solady/utils/EIP712.sol";
 ///
 /// @author Coinbase (https://github.com/coinbase/smart-wallet-permissions)
 contract SpendPermissionManager is EIP712 {
+    /// @notice A batch of spend permissions for an external spender to spend an account's tokens.
+    struct SpendPermissionBatch {
+        /// @dev Token addresses (ERC-7528 ether address or ERC-20 contract).
+        address[] tokens;
+        /// @dev Maximum allowed value to spend within a recurring period per token.
+        uint160[] allowances;
```

Comment:
Separate arrays have less syntax which is nice, but also places a responsibility of enforcing length equivalence checks. Separate arrays also feel a bit less intuitive for offchain preparation imo which I would index more on. What do you think?